mlir-sycl/test/Conversion/SYCLToLLVM/sycl-methods-to-llvm-typed-pointer.mlir

```diff
@@ -1333,21 +1333,29 @@ module attributes {gpu.container} {
 // CHECK-SAME:                    %[[VAL_1:.*]]: i32) -> i64 {
 // CHECK:             %[[VAL_2:.*]] = llvm.mlir.addressof @__spirv_BuiltInLocalInvocationId : !llvm.ptr<vector<3xi64>, 1>
 // CHECK:             %[[VAL_3:.*]] = llvm.load %[[VAL_2]] : !llvm.ptr<vector<3xi64>, 1>
-// CHECK-DAG:         %[[VAL_4:.*]] = llvm.mlir.constant(dense<0> : vector<3xindex>) : vector<3xi64>
-// CHECK-DAG:         %[[VAL_5:.*]] = llvm.mlir.constant(0 : i32) : i32
-// CHECK:             %[[VAL_6:.*]] = llvm.extractelement %[[VAL_3]]{{\[}}%[[VAL_5]] : i32] : vector<3xi64>
-// CHECK:             %[[VAL_7:.*]] = llvm.mlir.constant(0 : i64) : i64
-// CHECK:             %[[VAL_8:.*]] = llvm.insertelement %[[VAL_6]], %[[VAL_4]]{{\[}}%[[VAL_7]] : i64] : vector<3xi64>
-// CHECK:             %[[VAL_9:.*]] = llvm.mlir.constant(1 : i32) : i32
-// CHECK:             %[[VAL_10:.*]] = llvm.extractelement %[[VAL_3]]{{\[}}%[[VAL_9]] : i32] : vector<3xi64>
-// CHECK:             %[[VAL_11:.*]] = llvm.mlir.constant(1 : i64) : i64
-// CHECK:             %[[VAL_12:.*]] = llvm.insertelement %[[VAL_10]], %[[VAL_8]]{{\[}}%[[VAL_11]] : i64] : vector<3xi64>
-// CHECK:             %[[VAL_13:.*]] = llvm.mlir.constant(2 : i32) : i32
-// CHECK:             %[[VAL_14:.*]] = llvm.extractelement %[[VAL_3]]{{\[}}%[[VAL_13]] : i32] : vector<3xi64>
-// CHECK:             %[[VAL_15:.*]] = llvm.mlir.constant(2 : i64) : i64
-// CHECK:             %[[VAL_16:.*]] = llvm.insertelement %[[VAL_14]], %[[VAL_12]]{{\[}}%[[VAL_15]] : i64] : vector<3xi64>
-// CHECK:             %[[VAL_17:.*]] = llvm.extractelement %[[VAL_16]]{{\[}}%[[VAL_1]] : i32] : vector<3xi64>
-// CHECK:             llvm.return %[[VAL_17]] : i64
+// CHECK-DAG:         %[[VAL_4:.*]] = llvm.mlir.null : !llvm.ptr<struct<"class.sycl::_V1::id.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>>
+// CHECK-DAG:         %[[VAL_5:.*]] = llvm.mlir.constant(1 : index) : i64
+// CHECK:             %[[VAL_6:.*]] = llvm.getelementptr %[[VAL_4]]{{\[}}%[[VAL_5]]] : (!llvm.ptr<struct<"class.sycl::_V1::id.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>>, i64) -> !llvm.ptr<struct<"class.sycl::_V1::id.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>>
```

Comment:
Should we really include the `detail::` types in the check? In most other places, we have replace the part after the SYCL type (in this case `sycl::_V1::id.1`) with `{{.*}}` to make the tests more robust. 

Applies to multiple lines in both tests that were changed.

Reply:
Good point. Fixed!